AK/DOSPackedTime.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022, Undefine <undefine@undefine.pl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Time.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+union DOSPackedTime {
+    u16 value;
+    struct {
+        u16 second : 5;
+        u16 minute : 6;
+        u16 hour : 5;
+    };
+};
+static_assert(sizeof(DOSPackedTime) == 2);
+
+union DOSPackedDate {
+    u16 value;
+    struct {
+        u16 day : 5;
+        u16 month : 4;
+        u16 year : 7;
+    };
+};
+static_assert(sizeof(DOSPackedDate) == 2);
+
+inline constexpr u16 first_dos_year = 1980;
+
+Time time_from_packed_dos(DOSPackedDate, DOSPackedTime);
+DOSPackedDate to_packed_dos_date(unsigned, unsigned, unsigned);
+DOSPackedTime to_packed_dos_time(unsigned, unsigned, unsigned);
```

Comment:
Please name your parameters here, since it's not obvious from the types which is which.